controlplane/kubeadm/api/v1alpha3/conversion_test.go

```diff
@@ -37,28 +38,35 @@ func TestFuzzyConversion(t *testing.T) {
 	g.Expect(v1alpha4.AddToScheme(scheme)).To(Succeed())
 
 	t.Run("for KubeadmControlPLane", utilconversion.FuzzTestFunc(utilconversion.FuzzTestFuncInput{
-		Scheme: scheme,
-		Hub:    &v1alpha4.KubeadmControlPlane{},
-		Spoke:  &KubeadmControlPlane{},
-		FuzzerFuncs: []fuzzer.FuzzerFuncs{
-			func(codecs runtimeserializer.CodecFactory) []interface{} {
-				return []interface{}{
-					// This custom function is needed when ConvertTo/ConvertFrom functions
-					// uses the json package to unmarshal the bootstrap token string.
-					//
-					// The Kubeadm v1beta1.BootstrapTokenString type ships with a custom
-					// json string representation, in particular it supplies a customized
-					// UnmarshalJSON function that can return an error if the string
-					// isn't in the correct form.
-					//
-					// This function effectively disables any fuzzing for the token by setting
-					// the values for ID and Secret to working alphanumeric values.
-					func(in *kubeadmv1.BootstrapTokenString, c fuzz.Continue) {
-						in.ID = "abcdef"
-						in.Secret = "abcdef0123456789"
-					},
-				}
-			},
-		},
+		Scheme:      scheme,
+		Hub:         &v1alpha4.KubeadmControlPlane{},
+		Spoke:       &KubeadmControlPlane{},
+		FuzzerFuncs: []fuzzer.FuzzerFuncs{fuzzFuncs},
 	}))
 }
+
+func fuzzFuncs(_ runtimeserializer.CodecFactory) []interface{} {
+	// This custom function is needed when ConvertTo/ConvertFrom functions
+	// uses the json package to unmarshal the bootstrap token string.
+	//
+	// The Kubeadm v1beta1.BootstrapTokenString type ships with a custom
+	// json string representation, in particular it supplies a customized
+	// UnmarshalJSON function that can return an error if the string
+	// isn't in the correct form.
+	//
+	// This function effectively disables any fuzzing for the token by setting
+	// the values for ID and Secret to working alphanumeric values.
+	return []interface{}{
+		kubeadmBootstrapTokenStringFuzzer,
+		cabpkBootstrapTokenStringFuzzer,
+	}
+}
```

Comment:
This new split seems more complicated than having it inlined, could we have it like it was before?

Reply:
I have changed this for consistency with all the other fuzz functions implemented in other packages, but I have no strong opinions and I'm happy to revert it if you think this should be a special case